client_code/hashlib.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: MIT
+#
+# Copyright (c) 2021 The Anvil Extras project team members listed at
+# https://github.com/anvilistas/anvil-extras/graphs/contributors
+#
+# This software is published at https://github.com/anvilistas/anvil-extras
+
+from functools import partial
+
+from anvil.js import window
+
+__version__ = "2.0.1"
+
+
+def digest(algorithm, data):
+    """Returns the digest of the data using the specified algorithm.
+
+    Parameters
+    ----------
+    algorithm : str
+        The algorithm to use.
+    data : object
+        The data to digest.
+    """
+    if not isinstance(data, (bytes, str)):
+        data = str(data).encode("utf-8")
+
+    if isinstance(data, str):
+        data = data.encode("utf-8")
```

Comment:
why do we allow anything other than a str or bytes here?
just curious - seems odd to support a mutable data structure?